### DISCUSSION TITLE
two-sided marketplace

### DISCUSSION THREAD

**Discussion by erosinlove** [solidusio/solidus#4413] [New Features or Ideas]
I want to create a **two-sided marketplace** like Etsy, eBay, Uber, etc. with **solidus**. Can anyone advise or provide the guideline on how can we customize the solidus or what the solidus extension can create the two-sided marketplace?

**Comment by waiting-for-dev** [comment]
You could use https://github.com/solidusio-contrib/solidus_paypal_marketplace as starting point, but sure you'll need a good amount of customization. We're here to support you with any issues you might find in the process.

**Reply by erosinlove** [reply]
I created the app on macOS with the below gems

```
ruby "3.1.2"
gem "rails", "~> 7.0.3"
gem "pg", "~> 1.3", '>= 1.3.5'
gem 'solidus', github: 'solidusio/solidus'
gem "solidus_core", "~> 3.2.alpha"
gem "solidus_api", "~> 3.2.alpha"
gem 'solidus_auth_devise', '~> 2.5', '>= 2.5.4' 
```

I can run the sample store

but when I add PayPal
`gem 'solidus_paypal_commerce_platform' `

and run the installation generator
`bin/rails generate solidus_paypal_commerce_platform:install`

I got this error
```
/Users/xxx/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/solidus_paypal_commerce_platform-0.3.2/lib/solidus_paypal_commerce_platform/engine.rb:18:in `block in <class:Engine>': uninitialized constant SolidusPaypalCommercePlatform::PaymentMethod (NameError)

      app.config.spree.payment_methods << SolidusPaypalCommercePlatform::PaymentMethod
```

Could you please advise how can I fix this?

**Reply by waiting-for-dev** [reply]
That error is fixed on `solidus_paypal_commerce_platform v0.4`, recently released. Try to update it on your application: `bundle update --source solidus_paypal_commerce_platform`
